sonic-utilities-tests/config_mgmt_test.py

```diff
@@ -0,0 +1,475 @@
+        return
+
+    def test_search_keys(self):
+        iConfig = dict(configDbJson)
+        self.writeJson(iConfig, config_mgmt.CONFIG_DB_JSON_FILE)
+        cmdpb = config_mgmt.ConfigMgmtDPB(source=config_mgmt.CONFIG_DB_JSON_FILE)
+        out = cmdpb.configWithKeys(portBreakOutConfigDbJson, \
+            ["Ethernet8","Ethernet9"])
+        assert "VLAN" not in out.keys()
+        assert "INTERFACE" not in out.keys()
+        for k in out['ACL_TABLE'].keys():
+            # only ports must be chosen
+            len(out['ACL_TABLE'][k]) == 1
+        out = cmdpb.configWithKeys(portBreakOutConfigDbJson, \
+            ["Ethernet10","Ethernet11"])
+        assert "INTERFACE" in out.keys()
+        for k in out['ACL_TABLE'].keys():
+            # only ports must be chosen
+            len(out['ACL_TABLE'][k]) == 1
+        return
+
+    def test_break_out(self):
+        # prepare default config
+        self.writeJson(portBreakOutConfigDbJson, \
+            config_mgmt.DEFAULT_CONFIG_DB_JSON_FILE)
+        # prepare config dj json to start with
+        iConfig = dict(configDbJson)
+        #Ethernet8: start from 4x25G-->2x50G with -f -l
+        self.dpb_port8_4x25G_2x50G_f_l(iConfig)
+        #Ethernet8: move from 2x50G-->1x100G without force, list deps
+        self.dpb_port8_2x50G_1x100G(iConfig)
+        # Ethernet8: move from 2x50G-->1x100G with force, where deps exists
+        self.dpb_port8_2x50G_1x100G_f(iConfig)
+        # Ethernet8: move from 1x100G-->4x25G without force, no deps
+        self.dpb_port8_1x100G_4x25G(iConfig)
+        # Ethernet8: move from 4x25G-->1x100G with force, no deps
+        self.dpb_port8_4x25G_1x100G_f(iConfig)
+        # Ethernet8: move from 1x100G-->1x50G(2)+2x25G(2) with -f -l,
+        self.dpb_port8_1x100G_1x50G_2x25G_f_l(iConfig)
+        # Ethernet4: breakout from 4x25G to 2x50G with -f -l
+        self.dpb_port4_4x25G_2x50G_f_l(iConfig)
+        return
+
+    def tearDown(self):
+        try:
+            os.remove(config_mgmt.CONFIG_DB_JSON_FILE)
+            os.remove(config_mgmt.DEFAULT_CONFIG_DB_JSON_FILE)
+        except Exception as e:
+            pass
+        return
+
+    ########### HELPER FUNCS #####################################
+    def writeJson(self, d, file):
+        with open(file, 'w') as f:
+            dump(d, f, indent=4)
+        return
+    # config_mgmt.ConfigMgmtDPB class instance with mocked functions. Not using
+    # pytest fixture, because it is used in non test funcs.
+    def config_mgmt_dpb(self, iConfig):
+        # create object
+        self.writeJson(iConfig, config_mgmt.CONFIG_DB_JSON_FILE)
+        cmdpb = config_mgmt.ConfigMgmtDPB(source=config_mgmt.CONFIG_DB_JSON_FILE)
+        # mock funcs
+        cmdpb.writeConfigDB = MagicMock(return_value=True)
+        cmdpb._verifyAsicDB = MagicMock(return_value=True)
+        return cmdpb
+
+    # Generate port to deleted, added and lanes and speed setting based on
+    # current and new mode.
```

Comment:
Write one example

Reply:
Example: 
For generate_args(8, 73, '4x25G', '2x50G'):
output:
(
['Ethernet8', 'Ethernet9', 'Ethernet10', 'Ethernet11'], 
['Ethernet8', 'Ethernet10'], 
{'Ethernet8': {'lanes': '73,74', 'speed': '50000'}, 'Ethernet10': {'lanes': '75,76', 'speed': '50000'}})

For generate_args(8, 73, '1x100G, ''4x25G'):
output:
(['Ethernet8'], 
['Ethernet8', 'Ethernet9', 'Ethernet10', 'Ethernet11'], 
{'Ethernet8': {'lanes': '73', 'speed': '25000'}, 'Ethernet9': {'lanes': '74', 'speed': '25000'}, 'Ethernet10': {'lanes': '75', 'speed': '25000'}, 'Ethernet11': {'lanes': '76', 'speed': '25000'}})